### DISCUSSION TITLE
How do I force treesitters `ensured_installed` to be empty?

### DISCUSSION THREAD

**Discussion by UnderTheCarpet** [LazyVim/LazyVim#3702] [Q&A] [answered]
For older releasese this worked, but does not in `v12.13.0`.
```lua
 require("lazy").setup({
        ...
        spec = {
          { "LazyVim/LazyVim", import = "lazyvim.plugins" },
          ...
          { "nvim-treesitter/nvim-treesitter", opts = { ensure_installed = {} } },
        },
      })

```

In the above version (which is available to me), this option is populated regardless (which I check via `:lua print(vim.inspect(require("lazy").plugins()))`.

Use case: I install the parsers via nix.

**Comment by dpetka2001** [comment, marked as answer]
Try 
```lua
    opts = function(_, opts)
      opts.ensure_installed = {}
    end,
```

**Reply by UnderTheCarpet** [reply]
That seems to be working. Thx! Can you explain, how this works? How does this shortcut the merging of tables?

**Reply by dpetka2001** [reply]
When you use a function `opts` you're essentially setting the value of a key to whatever you want overwriting the default value (in your case just an empty table).

**Reply by diocletiann** [reply]
thanks for this!
